exchanges/orderbook/linked_list.go

```diff
@@ -172,10 +172,13 @@ func (ll *linkedList) amount() (liquidity, value float64) {
 }
 
 // retrieve returns a full slice of contents from the linked list
-func (ll *linkedList) retrieve() Items {
-	depth := make(Items, 0, ll.length)
-	for tip := ll.head; tip != nil; tip = tip.Next {
-		depth = append(depth, tip.Value)
+func (ll *linkedList) retrieve(count int) Items {
+	if count == 0 || ll.length < count {
+		count = ll.length
+	}
+	depth := make(Items, count)
+	for i, tip := 0, ll.head; i < count; i, tip = i+1, tip.Next {
+		depth[i] = tip.Value
```

Comment:
This is an observation. The previous work does nil checks despite using the `ll.length` property too. The only question I raise for you to consider is: has anything changed to make you trust the length more to drop the nil check and introduce a potential panic? Are you comfortable with such a possibility deep in the depths?

Reply:
🦅